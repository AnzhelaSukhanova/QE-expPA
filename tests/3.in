(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and (and (or (bvult y (bvmul x #b0110)) (= y (bvmul x #b0110))) (or (bvult y (bvmul x #b0011)) (= y (bvmul x #b0011)))) (or (bvult y (bvmul x #b0010)) (= y (bvmul x #b0010))))))
