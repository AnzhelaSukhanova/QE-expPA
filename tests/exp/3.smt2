(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (or (bvult (bvadd (bvshl #b0001 x) x) (bvadd (bvadd (bvshl #b0001 y) y) #b0001)) (= (bvadd (bvshl #b0001 x) x) (bvadd (bvadd (bvshl #b0001 y) y) #b0001)))))
