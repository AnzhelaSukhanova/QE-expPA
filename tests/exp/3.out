(let (($e1 (bvshl #b0101 y))) (let (($e2 (bvadd $e1 #b0111))) (let (($e3 (ite (bvult $e2 #b1000) (ite (bvult $e2 #b0100) (ite (bvult $e2 #b0010) (ite (= $e1 #b1001) #b0100 #b0000) #b0001) #b0010) #b0011))) (let (($e4 (bvshl #b0001 $e3))) (let (($e5 (bvshl #b1010 y))) (let (($e6 (bvadd #b1000 $e5))) (let (($e7 (bvshl #b0001 (bvadd #b0001 $e3)))) (let (($e8 (= #b1000 $e5))) (not (and (not (= #b1010 $e5)) (not (= #b1001 $e5)) (not (bvult #b0010 $e6)) (not (bvult #b0001 $e6)) (not (= #b1100 $e5)) (not (bvult #b0100 $e6)) (not (= #b0000 $e5)) (not (bvult #b1000 $e6)) (not (= $e6 $e7)) (not (= $e4 $e6)) (not $e8) (not (bvult $e7 $e6)) (not (bvult $e4 $e6)) $e8))))))))))
